### PR TITLE
[RUMF-1028] enable privacy by default

### PR DESCRIPTION
**Pull request by BenoitZugmeyer**
## Motivation

Let our users specify a initial privacy level for session replay. This should default to `mask-forms-only`.

## Changes

Enable privacy by default, in particular the `initialPrivacyLevel` option, and change its default.

## Testing

CI + tested on org2

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by acorretti — APPROVED** [review]

File: packages/rum/src/boot/startRecording.spec.ts [hunk 1]
```diff
@@ -52,6 +52,9 @@ describe('startRecording', () => {
         },
       })
       .withSession(session)
+      .withConfiguration({
+        initialPrivacyLevel: InitialPrivacyLevel.ALLOW,
```
Comment (BenoitZugmeyer) [inline packages/rum/src/boot/startRecording.spec.ts:56]: With the default privacy level (mask forms only) data compression is too efficient (🎉), so segments are not reaching the compressed size limit.

**Review by acorretti — CHANGES_REQUESTED** [review]
Let's rephrase `MASK_FORMS_ONLY` to `MASK_USER_INPUT`
And also `InitialPrivacyLevel` to `DefaultPrivacyLevel`

Has been discussed with PM already

**Comment by jagracey** [comment]
@acorretti To be clear here, regarding the public facing "privacy" API spec, you mean the following changes?

**Exhaustively (and redundantly) the changes are:**
1. The JS config property name currently `initialPrivacyLevel` to `defaultPrivacyLevel `.
2. The JS config property value currently `mask-forms-only` to `mask-user-input`.
3. The privacy attribute tag value from `mask-forms-only` to `mask-user-input`.
4. The privacy class name from `dd-privacy-mask-forms-only` to `dd-privacy-mask-user-input`

**Comment by acorretti** [comment]
That'd be it @jagracey 👍

**Comment by jagracey** [comment]
LGTM

**Review by acorretti — APPROVED** [review]

**Review by jagracey — APPROVED** [review]

File: packages/core/src/domain/configuration/configuration.ts [hunk 2]
```diff
@@ -4,12 +4,12 @@ import { catchUserErrors } from '../../tools/catchUserErrors'
 import { includes, objectHasValue, ONE_KILO_BYTE, ONE_SECOND } from '../../tools/utils'
 import { computeTransportConfiguration, TransportConfiguration } from './transportConfiguration'
 
-export const InitialPrivacyLevel = {
+export const DefaultPrivacyLevel = {
   ALLOW: 'allow',
   MASK: 'mask',
-  MASK_FORMS_ONLY: 'mask-forms-only',
+  MASK_USER_INPUT: 'mask-user-input',
```
Comment (amortemousque) [inline packages/core/src/domain/configuration/configuration.ts:10]: Shouldn't it be plural 'mask-user-inputs'

Comment (acorretti) [reply packages/core/src/domain/configuration/configuration.ts:10]: I don't really think so, as it's not talking about the `<input>` elements but rather user input as a concept of recorded data.

**Review by amortemousque — APPROVED** [review]

**Review by acorretti — APPROVED** [review]

**Comment by eddiemonge** [comment]
This is missing updated docs for https://docs.datadoghq.com/real_user_monitoring/browser

**Comment by bcaudan** [comment]
@eddiemonge are you looking for [session replay privacy options doc](https://docs.datadoghq.com/real_user_monitoring/session_replay/privacy_options/?tab=maskuserinput)?

**Comment by eddiemonge** [comment]
@bcaudan its nice that its documented there, but it should also be part of the configuration initialization docs. https://docs.datadoghq.com/real_user_monitoring/browser/#configuration

**Comment by bcaudan** [comment]
Indeed, it should land soon, cf https://github.com/DataDog/browser-sdk/pull/1152

**Pull request merged**